### ISSUE TITLE
no output is created

### ISSUE THREAD

**Issue by monajalal** [TadasBaltrusaitis/OpenFace#299]
Hi, I am trying to run the script on 30k images however no result is created:

```
root@4d2d2ef45ab8:/opt/OpenFace/build/bin# **./FaceLandmarkImg -fdir "/opt/OpenFace/build/bin/images/" -ofir "/opt/OpenFace/build/bin/output_aus_30k/"**
Loading the model
Reading the CLNF landmark detector/tracker from: model/main_clnf_general.txt
Reading the landmark detector module from: model/clnf_general.txt
Reading the PDM module from: model/pdms/In-the-wild_aligned_PDM_68.txt....Done
Reading the Triangulations module from: model/tris_68.txt....Done
Reading the intensity CCNF patch experts from: model/patch_experts/ccnf_patches_0.25_general.txt....Done
Reading the intensity CCNF patch experts from: model/patch_experts/ccnf_patches_0.35_general.txt....Done
Reading the intensity CCNF patch experts from: model/patch_experts/ccnf_patches_0.5_general.txt....Done
Reading part based module....inner
Reading the CLNF landmark detector/tracker from: model/model_inner/main_clnf_inner.txt
Reading the landmark detector module from: model/model_inner/clnf_inner.txt
Reading the PDM module from: model/model_inner/pdms/pdm_51_inner.txt....Done
Reading the intensity CCNF patch experts from: model/model_inner/patch_experts/ccnf_patches_1.00_inner.txt....Done
Done
Reading part based module....left_eye_28
Reading the CLNF landmark detector/tracker from: model/model_eye/main_clnf_synth_left.txt
Reading the landmark detector module from: model/model_eye/clnf_left_synth.txt
Reading the PDM module from: model/model_eye/pdms/pdm_28_l_eye_3D_closed.txt....Done
Reading the intensity CCNF patch experts from: model/model_eye/patch_experts/left_ccnf_patches_1.00_synth_lid_.txt....Done
Reading the intensity CCNF patch experts from: model/model_eye/patch_experts/left_ccnf_patches_1.50_synth_lid_.txt....Done
Done
Reading part based module....right_eye_28
Reading the CLNF landmark detector/tracker from: model/model_eye/main_clnf_synth_right.txt
Reading the landmark detector module from: model/model_eye/clnf_right_synth.txt
Reading the PDM module from: model/model_eye/pdms/pdm_28_eye_3D_closed.txt....Done
Reading the intensity CCNF patch experts from: model/model_eye/patch_experts/ccnf_patches_1.00_synth_lid_.txt....Done
Reading the intensity CCNF patch experts from: model/model_eye/patch_experts/ccnf_patches_1.50_synth_lid_.txt....Done
Done
Reading the landmark validation module....Done
Model loaded

(colour:48): Gtk-WARNING **: cannot open display: localhost:14.0
root@4d2d2ef45ab8:/opt/OpenFace/build/bin# ls output_aus_30k/
root@4d2d2ef45ab8:/opt/OpenFace/build/bin# **./FaceLandmarkImg -fdir /opt/OpenFace/build/bin/images/ -ofir /opt/OpenFace/build/bin/output_aus_30k/** 
Loading the model
Reading the CLNF landmark detector/tracker from: model/main_clnf_general.txt
Reading the landmark detector module from: model/clnf_general.txt
Reading the PDM module from: model/pdms/In-the-wild_aligned_PDM_68.txt....Done
Reading the Triangulations module from: model/tris_68.txt....Done
Reading the intensity CCNF patch experts from: model/patch_experts/ccnf_patches_0.25_general.txt....Done
Reading the intensity CCNF patch experts from: model/patch_experts/ccnf_patches_0.35_general.txt....Done
Reading the intensity CCNF patch experts from: model/patch_experts/ccnf_patches_0.5_general.txt....Done
Reading part based module....inner
Reading the CLNF landmark detector/tracker from: model/model_inner/main_clnf_inner.txt
Reading the landmark detector module from: model/model_inner/clnf_inner.txt
Reading the PDM module from: model/model_inner/pdms/pdm_51_inner.txt....Done
Reading the intensity CCNF patch experts from: model/model_inner/patch_experts/ccnf_patches_1.00_inner.txt....Done
Done
Reading part based module....left_eye_28
Reading the CLNF landmark detector/tracker from: model/model_eye/main_clnf_synth_left.txt
Reading the landmark detector module from: model/model_eye/clnf_left_synth.txt
Reading the PDM module from: model/model_eye/pdms/pdm_28_l_eye_3D_closed.txt....Done
Reading the intensity CCNF patch experts from: model/model_eye/patch_experts/left_ccnf_patches_1.00_synth_lid_.txt....Done
Reading the intensity CCNF patch experts from: model/model_eye/patch_experts/left_ccnf_patches_1.50_synth_lid_.txt....Done
Done
Reading part based module....right_eye_28
Reading the CLNF landmark detector/tracker from: model/model_eye/main_clnf_synth_right.txt
Reading the landmark detector module from: model/model_eye/clnf_right_synth.txt
Reading the PDM module from: model/model_eye/pdms/pdm_28_eye_3D_closed.txt....Done
Reading the intensity CCNF patch experts from: model/model_eye/patch_experts/ccnf_patches_1.00_synth_lid_.txt....Done
Reading the intensity CCNF patch experts from: model/model_eye/patch_experts/ccnf_patches_1.50_synth_lid_.txt....Done
Done
Reading the landmark validation module....Done
Model loaded

(colour:61): Gtk-WARNING **: cannot open display: localhost:14.0
root@4d2d2ef45ab8:/opt/OpenFace/build/bin# 

```

I started the docker this way:
` sudo docker pull benbuleong/openface-cambridge`

```
$ xhost +
```
```
$ sudo docker run -p 9000:9000 -p 8000:8000  -e DISPLAY=$DISPLAY -v /tmp/.X11-unix:/tmp/.X11-unix -v /scratch2/datasets/new_aff/manually/Manually_Annotated_Images/30k_images:/opt/OpenFace/build/bin/images:z -t -i benbuleong/openface-cambridge /bin/bash
```

Any help is really appreciated.

**Comment by monajalal** [comment]
Additionally,

When I run the `FaceLandmarkImg` script within the docker, it does create output only for the first image and then stops creating outputs.
```

    [jalal@goku ~]$ sudo docker pull benbuleong/openface-cambridge
    [jalal@goku ~]$ xhost + 
    access control disabled, clients can connect from any host
    [jalal@goku ~]$ sudo docker run -p 9000:9000 -p 8000:8000  -e DISPLAY=$DISPLAY -v /tmp/.X11-unix:/tmp/.X11-unix -v /scratch2/datasets/new_aff/manually/Manually_Annotated_Images/30k_images:/opt/OpenFace/build/bin/images:z -t -i benbuleong/openface-cambridge /bin/bash 
    root@5b37ee4cd1b6:/opt/OpenFace/build/bin#  ./FaceLandmarkImg -fdir /opt/OpenFace/build/bin/images/ -ofdir /opt/OpenFace/build/bin/output_aus_30k/ 
    Failed to create a directory... /opt/OpenFace/build/bin/output_aus_30k/
    Loading the model
    Reading the CLNF landmark detector/tracker from: model/main_clnf_general.txt
    Reading the landmark detector module from: model/clnf_general.txt
    Reading the PDM module from: model/pdms/In-the-wild_aligned_PDM_68.txt....Done
    Reading the Triangulations module from: model/tris_68.txt....Done
    Reading the intensity CCNF patch experts from: model/patch_experts/ccnf_patches_0.25_general.txt....Done
    Reading the intensity CCNF patch experts from: model/patch_experts/ccnf_patches_0.35_general.txt....Done
    Reading the intensity CCNF patch experts from: model/patch_experts/ccnf_patches_0.5_general.txt....Done
    Reading part based module....inner
    Reading the CLNF landmark detector/tracker from: model/model_inner/main_clnf_inner.txt
    Reading the landmark detector module from: model/model_inner/clnf_inner.txt
    Reading the PDM module from: model/model_inner/pdms/pdm_51_inner.txt....Done
    Reading the intensity CCNF patch experts from: model/model_inner/patch_experts/ccnf_patches_1.00_inner.txt....Done
    Done
    Reading part based module....left_eye_28
    Reading the CLNF landmark detector/tracker from: model/model_eye/main_clnf_synth_left.txt
    Reading the landmark detector module from: model/model_eye/clnf_left_synth.txt
    Reading the PDM module from: model/model_eye/pdms/pdm_28_l_eye_3D_closed.txt....Done
    Reading the intensity CCNF patch experts from: model/model_eye/patch_experts/left_ccnf_patches_1.00_synth_lid_.txt....Done
    Reading the intensity CCNF patch experts from: model/model_eye/patch_experts/left_ccnf_patches_1.50_synth_lid_.txt....Done
    Done
    Reading part based module....right_eye_28
    Reading the CLNF landmark detector/tracker from: model/model_eye/main_clnf_synth_right.txt
    Reading the landmark detector module from: model/model_eye/clnf_right_synth.txt
    Reading the PDM module from: model/model_eye/pdms/pdm_28_eye_3D_closed.txt....Done
    Reading the intensity CCNF patch experts from: model/model_eye/patch_experts/ccnf_patches_1.00_synth_lid_.txt....Done
    Reading the intensity CCNF patch experts from: model/model_eye/patch_experts/ccnf_patches_1.50_synth_lid_.txt....Done
    Done
    Reading the landmark validation module....Done
    Model loaded
    
    (colour:16): Gtk-WARNING **: cannot open display: :0
    root@5b37ee4cd1b6:/opt/OpenFace/build/bin# 

```

As you can see here, the FaceLandmarkImg script works for the first image and creates the related result and then somehow gets stopped. 
     
 ```
   root@5b37ee4cd1b6:/opt/OpenFace/build/bin# ls           
    AU_predictors  FaceLandmarkImg  FaceLandmarkVid  FaceLandmarkVidMulti  FeatureExtraction  classifiers  images  model  output_aus_30k
    root@5b37ee4cd1b6:/opt/OpenFace/build/bin# cd output_aus_30k/
    root@5b37ee4cd1b6:/opt/OpenFace/build/bin/output_aus_30k# ls
    0000792211b64f6a59bd2d95fee49eabe6373ec1d88f2285bae2ba9b_det_0.pts
    root@5b37ee4cd1b6:/opt/OpenFace/build/bin/output_aus_30k# cd ../images/
    root@5b37ee4cd1b6:/opt/OpenFace/build/bin/images# ls | head -1
    0000792211b64f6a59bd2d95fee49eabe6373ec1d88f2285bae2ba9b.jpg
    root@5b37ee4cd1b6:/opt/OpenFace/build/bin/images# ls -1 | wc -l
    30000
    root@5b37ee4cd1b6:/opt/OpenFace/build/bin/images# cd ..
    root@5b37ee4cd1b6:/opt/OpenFace/build/bin# cd output_aus_30k/
    root@5b37ee4cd1b6:/opt/OpenFace/build/bin/output_aus_30k# ls
    0000792211b64f6a59bd2d95fee49eabe6373ec1d88f2285bae2ba9b_det_0.pts
    
```
    
    Additionally, I ran it with this other option I found on Web:
`    root@5b37ee4cd1b6:/opt/OpenFace/build/bin# ./FaceLandmarkImg -fdir "./images/" -ofdir "./output_aus_30k/" -oidir "./output_aus_30k/" -wild`
Still no luck.

I am following the command line instruction from OpenFace official Website https://github.com/TadasBaltrusaitis/OpenFace/wiki/Command-line-arguments#facelandmarkimg 

My docker version is:
```

    [jalal@goku ~]$ docker -v
    Docker version 1.12.6, build 85d7426/1.12.6

```
Here's my docker info:

```
    [sudo] password for jalal: 
    Containers: 26
     Running: 1
     Paused: 0
     Stopped: 25
    Images: 2
    Server Version: 1.12.6
    Storage Driver: devicemapper
     Pool Name: docker-253:0-38314914-pool
     Pool Blocksize: 65.54 kB
     Base Device Size: 10.74 GB
     Backing Filesystem: xfs
     Data file: /dev/loop0
     Metadata file: /dev/loop1
     Data Space Used: 12.33 GB
     Data Space Total: 107.4 GB
     Data Space Available: 9.965 GB
     Metadata Space Used: 15.06 MB
     Metadata Space Total: 2.147 GB
     Metadata Space Available: 2.132 GB
     Thin Pool Minimum Free Space: 10.74 GB
     Udev Sync Supported: true
     Deferred Removal Enabled: true
     Deferred Deletion Enabled: true
     Deferred Deleted Device Count: 0
     Data loop file: /var/lib/docker/devicemapper/devicemapper/data
     WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
     Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
     Library Version: 1.02.140-RHEL7 (2017-05-03)
    Logging Driver: none
    Cgroup Driver: systemd
    Plugins:
     Volume: local
     Network: bridge overlay null host
    Swarm: inactive
    Runtimes: docker-runc runc
    Default Runtime: docker-runc
    Security Options: seccomp selinux
    Kernel Version: 3.10.0-693.5.2.el7.x86_64
    Operating System: CentOS Linux 7 (Core)
    OSType: linux
    Architecture: x86_64
    Number of Docker Hooks: 3
    CPUs: 12
    Total Memory: 62.67 GiB
    Name: goku.bu.edu
    ID: E6DT:PP6P:WHVY:Z3NQ:N2JT:LJDV:EXOS:WG3V:72Z7:62ML:DW2Q:RY4Q
    Docker Root Dir: /var/lib/docker
    Debug Mode (client): false
    Debug Mode (server): false
    Registry: https://index.docker.io/v1/
    Insecure Registries:
     127.0.0.0/8
    Registries: docker.io (secure)
```

**Comment by NumesSanguis** [comment]
This is how I got the docker version to work:
http://jinhyuncheong.com/jekyll/update/2017/08/03/Using-OpenFace-Cambridge-Face-Expression-Toolbox.html

But it took me a while to find these instructions. Is it possible to add some instructions for how to use Docker with OpenFace?

I was trying to get OpenFace to work with OpenCV 3.3 (because OpenCV 3.1 didn't want to install on my machine), and although the installation had no error, I also have no output file when using FaceLandmarkImg:
*my own issue*

I noticed that the Docker version however uses a different output format (it uses brackets), while newer versions have a .csv file structure. It would be much appreciated if you could update the docker version.

tl;dr: Include Docker instructions in the Wiki and update the Docker image to a more recent version.

**Comment by TadasBaltrusaitis** [comment]
Hi,

Thanks for the suggestions. Currently as it stands the Docker image was not created by me and is not officially supported. However, it would be great to add that as part of an official support, although not in the near future.

-Tadas
